core/src/main/scala/org/apache/spark/resource/ResourceDiscoveryScriptPlugin.scala

```diff
@@ -32,6 +33,7 @@ import org.apache.spark.util.Utils.executeAndGetOutput
  * If the user specifies custom plugins, this is the last one to be executed and
  * throws if the resource isn't discovered.
  */
+@DeveloperApi
 class ResourceDiscoveryScriptPlugin extends ResourceDiscoveryPlugin with Logging {
```

Comment:
I just happened to double check these modules. Shall we add `@since` for APIs `ResourceDiscoveryScriptPlugin`, `ResourceInformation`, etc. ?

I remember it was targeted to be APIs for Spark 3.0, and we kind of postponed to Spark 3.1 (?) cc @tgravescs - am I correct? Let's add `@since` for exposed APIs for Spark 3.0 at least.

Reply:
this is 3.0 and is a separate feature from the stage level scheduling that got postponed to 3.1.  We can add the @since and its 3.0